File: design-hashmap/design-hashmap.py
        self.arr[idx].append([key, value])
​
    def get(self, key: int) -> int:
        """
        Returns the value to which the specified key is mapped, or -1 if this map contains no mapping for the key
        """
        idx = key % 769
        
        for k, v in self.arr[idx]:
            if k == key:
                return v
        
        return -1
​
    def remove(self, key: int) -> None:
        """
        Removes the mapping of the specified value key if this map contains a mapping for the key
        """
        idx = key % 769
        
        for i in range(len(self.arr[idx])):
            k, v = self.arr[idx][i]
            if k == key:
                self.arr[idx][i] = self.arr[idx][len(self.arr[idx]) - 1]
                self.arr[idx] = self.arr[idx][:-1]
                return
​
​
# Your MyHashMap object will be instantiated and called as such:
# obj = MyHashMap()
# obj.put(key,value)
# param_2 = obj.get(key)
# obj.remove(key)

File: design-hashmap/design-hashmap.java
class MyHashMap {
    List<List<int[]>> arr;
        
    /** Initialize your data structure here. */
    public MyHashMap() {
        arr = new ArrayList<List<int[]>>();
        for (int i = 0; i < 769; i++) {
            arr.add(new ArrayList<int[]>());
        }
    }
    
    /** value will always be non-negative. */
    public void put(int key, int value) {
        int idx = key % 769;
        List<int[]> bucket = arr.get(idx);
        
        for (int i = 0; i < bucket.size(); i++) {
            int[] pairs = bucket.get(i);
            if (pairs[0] == key) {
                pairs[1] = value;
                bucket.set(i, pairs);
                arr.set(idx, bucket);
                return;
            }
        }
        
        int[] add = new int[]{key, value};
        bucket.add(add);
        arr.set(idx, bucket);
    }
    
    /** Returns the value to which the specified key is mapped, or -1 if this map contains no mapping for the key */
    public int get(int key) {
        int idx = key % 769;
        List<int[]> bucket = arr.get(idx);
        
        for (int i = 0; i < bucket.size(); i++) {
            int[] pairs = bucket.get(i);
            if (pairs[0] == key) {
                return pairs[1];
            }
        }
        
        return -1;
    }
    
    /** Removes the mapping of the specified value key if this map contains a mapping for the key */
    public void remove(int key) {
        int idx = key % 769;
        List<int[]> bucket = arr.get(idx);
        
        for (int i = 0; i < bucket.size(); i++) {
            int[] pairs = bucket.get(i);
            if (pairs[0] == key) {
                bucket.set(i, bucket.get(bucket.size() - 1));
                bucket.remove(bucket.size() - 1);
            }
        }
    }
}

/**
 * Your MyHashMap object will be instantiated and called as such:
 * MyHashMap obj = new MyHashMap();
 * obj.put(key,value);
 * int param_2 = obj.get(key);
 * obj.remove(key);
 */